### PR TITLE
feat: extend session lifespan on session refresh (skip=true)

### PR DESCRIPTION
**Pull request by sgal**
## Related issue(s)
Picking up work done in https://github.com/ory/hydra/pull/2848 to speed up the delivery, since the author is on vacation.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: persistence/sql/migrations/20220914000001000000_support_refresh_remember_for.sqlite.up.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE hydra_oauth2_flow ADD login_extend_session_lifespan BOOLEAN NOT NULL DEFAULT FALSE;
```
Comment (aeneasr) [inline persistence/sql/migrations/20220914000001000000_support_refresh_remember_for.sqlite.up.sql:1]: The migration timestamp should be later than the last migration which is:

```
20230228000010000001_client_add_skip_consent_column.up
```

Maybe just use the current date:

```
20230313112801000001_support_refresh_remember_for
```

File: consent/types.go [hunk 2]
```diff
@@ -265,6 +265,10 @@ type HandledLoginRequest struct {
 	// authorization will be remembered for the duration of the browser session (using a session cookie).
 	RememberFor int `json:"remember_for"`
 
+	// ExtendSessionLifespan, if set to true, session cookie expiry time will be updated when session is
+	// refreshed (login skip=true).
+	ExtendSessionLifespan bool `json:"extend_session_lifespan"`
```
Comment (aeneasr) [inline consent/types.go:270]:
```suggestion
	// Extend OAuth2 authentication session lifespan
	//
	// If set to `true`, the OAuth2 authentication cookie lifespan is extended. This is for example useful if you want the user to be able to use `prompt=none` continuously.
	//
	// This value can only be set to `true` if the user has an authentication, which is the case if the `skip` value is `true`.
	//
	// required: false
	ExtendSessionLifespan bool `json:"extend_session_lifespan"`
```

**Review by aeneasr — CHANGES_REQUESTED** [review]
Please add a migration test file for the new migration :)

**Comment by sgal** [comment]
@aeneasr Added migration test and fixed comments, please have a look!

**Review by aeneasr — APPROVED** [review]
Great job, thank you very much!

**Pull request merged**